react-ui/src/pages/contest/ContestGrid.jsx

```diff
@@ -0,0 +1,255 @@
+              <Card id={id} className={classes.entry}>
+                <CardContent className={classes.entryHeading}>
+                  {rank && (
+                    <Typography component="div" variant="h6">
+                      <span className={classes.numberSymbol}>#</span>
+                      {rank}
+                    </Typography>
+                  )}
+                  <div className={classes.entryInfo}>
+                    <Box alignItems="flex-start" display="flex" flexGrow={1}>
+                      <Box flexGrow={1}>
+                        <Typography component="div" variant="subtitle2">
+                          {entryName}
+                        </Typography>
+                        {user && (
+                          <Typography variant="caption">
+                            <RedditUserAttribution user={user} />
+                          </Typography>
+                        )}
+                      </Box>
+                      {/* <Experiment name="contest_card_description">
+                        <CustomIconButton
+                          ariaLabel="View description"
+                          className={classes.descriptionIcon}
+                          Icon={DescriptionIcon}
+                          onClick={() => {
+                            viewDescription(id);
+                          }}
+                          size="small"
+                        />
+                      </Experiment> */}
+                    </Box>
+                    {(!isContestMode || category) && (
+                      <div className={classes.entryRatings}>
+                        {category && (
+                          <CategoryLabel
+                            categories={categories}
+                            category={category}
+                            categoryRank={categoryRank}
+                          />
+                        )}
+                        {!isContestMode && (
+                          <>
+                            <Average average={average} fullText={rank === '1'} />
+                            {rating > -1 && (
+                              <Typography className={classes.myRating} variant="caption">
+                                {rank === '1' && <span>My&nbsp;rating:&nbsp;</span>}
+                                <FiveStar rating={rating} />
+                              </Typography>
+                            )}
+                          </>
+                        )}
+                      </div>
+                    )}
+                  </div>
+                </CardContent>
+                <div className={classes.entryImageContainer}>
+                  <CardImageLink
+                    displayWidth={rank === '1' ? winnerDisplayWidth : gridDisplayWidth}
+                    height={height}
+                    id={id}
+                    image={imagePath}
+                    nextState={{ selectedCategories }}
```

Comment:
Is `nextState` a react prop? I don't see it in `CardImageLink`.

Reply:
nextState was old code. You had passed selectedCategories into CardLink, which was then pulled from state to reselect them when you returned to the contests. I just forgot to remove it here.